cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -1746,30 +1707,36 @@ public void pushSubscriptionAttributeValue(String attributeId, String value) {
                 }
                 subscriptionAttributes.put(attributeId, arrayString);
 
-                CleverPushHttpClient.post("/subscription/attribute/push-value", jsonBody, new CleverPushHttpClient.ResponseHandler() {
-                    @Override
-                    public void onSuccess(String response) {
-                        try {
-                            if (sharedPreferences != null) {
-                                JSONObject jsonObject = new JSONObject(subscriptionAttributes);
-                                String jsonString = jsonObject.toString();
-                                SharedPreferences.Editor editor = sharedPreferences.edit();
-                                editor.remove(CleverPushPreferences.SUBSCRIPTION_ATTRIBUTES).apply();
-                                editor.putString(CleverPushPreferences.SUBSCRIPTION_ATTRIBUTES, jsonString);
-                                editor.commit();
-                            }
-                        } catch (Exception ex) {
-                            Log.e("CleverPush", ex.getMessage(), ex);
-                        }
-                    }
+                CleverPushHttpClient.post("/subscription/attribute/push-value", jsonBody, pushSubscriptionAttributeValueResponseHandler(subscriptionAttributes));
+            }
+        })).start());
+    }
 
-                    @Override
-                    public void onFailure(int statusCode, String response, Throwable throwable) {
-                        Log.e("CleverPush", "Error pushing attribute value - HTTP " + statusCode);
+    public CleverPushHttpClient.ResponseHandler pushSubscriptionAttributeValueResponseHandler(Map<String, Object> subscriptionAttributes) {
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(CleverPush.context);
+        CleverPushHttpClient.ResponseHandler responseHandler = new CleverPushHttpClient.ResponseHandler() {
```

Comment:
you can just return ResponseHandler directly here as well, no need for such a heavy variable to be reused in this context.